dashboards-observability/public/components/application_analytics/components/create.tsx

```diff
@@ -64,7 +64,7 @@ export const CreateApp = (props: CreateAppProps) => {
   const [selectedServices, setSelectedServices] = useState<OptionType[]>([]);
   const [selectedTraces, setSelectedTraces] = useState<OptionType[]>([]);
 
-  const editMode = !!existingAppId;
+  const editMode = existingAppId !== 'undefined';
```

Comment:
if existingAppId is `undefined` (not the string), then editMode is true but it's not actually edit mode? if so maybe `editMode = existingAppId != null && existingAppId !== 'undefined'` to be consistent with variable naming?

Reply:
Not exactly, existingAppId comes from `decodeURIComponent(routerProps.match.params.id)` which returns the string `"undefined"` if the parameter does not exist and the id when it does. So there should not be a case where the existingAppId is `undefined`(not the string)

Reply:
got it. then why `if (editMode && existingAppId)` instead of `if (editMode)`?